### PR TITLE
[infra] unit/integration test updates

### PR DESCRIPTION
**Pull request by Leo-Neat**
Updating some of the unit and integration tests for CIFuzz infrastructure 

NOTE: If you want to run build_specified_commit_test.py or bisector_test.py you need root privileges

### CODE REVIEW

File: infra/base-images/base-builder/detect_repo_test.py [hunk 1]
```diff
@@ -13,8 +13,10 @@
 # limitations under the License.
 """Test the functionality of the detect_repo module.
 This will consist of the following functional test:
-  1. Determine if a OSS-Fuzz projects main repo can be accurately deduce
-  from example commits.
+  1. Determine if an OSS-Fuzz projects main repo can be detected from example
```
Comment (Leo-Neat) [inline infra/base-images/base-builder/detect_repo_test.py]: project's

File: infra/base-images/base-builder/detect_repo_test.py [hunk 2]
```diff
@@ -13,8 +13,10 @@
...
+  commits.
+  2. Determine if an OSS-Fuzz project main repo can be detected from a
```
Comment (Leo-Neat) [inline infra/base-images/base-builder/detect_repo_test.py]: project's

File: infra/base-images/base-builder/detect_repo_test.py [hunk 3]
```diff
@@ -30,94 +32,53 @@
...
 class DetectRepoTest(unittest.TestCase):
   """Class to test the functionality of the detect_repo module."""
 
-  def test_infer_main_repo(self):
+  def test_infer_main_repo_from_commit(self):
     """Tests that the main repo can be inferred based on an example commit."""
 
     with tempfile.TemporaryDirectory() as tmp_dir:
-
       # Construct example repo's to check for commits.
```
Comment (Leo-Neat) [inline infra/base-images/base-builder/detect_repo_test.py]: Remove comment

File: infra/base-images/base-builder/detect_repo_test.py [hunk 4]
```diff
@@ -30,94 +32,53 @@
...
-      self.check_ref_with_repo('https://github.com/libarchive/libarchive.git',
-                               'libarchive', tmp_dir)
+      for example_repo in test_repos.TEST_REPOS:
+        repo_manager.RepoManager(example_repo.git_url, tmp_dir)
+        self.check_with_repo(example_repo.git_url, example_repo.git_repo_name,
+                             tmp_dir)
 
-  def check_ref_with_repo(self, repo_origin, repo_name, tmp_dir):
+  def check_with_repo(self, repo_origin, repo_name, tmp_dir, commit=None):
     """Checks the detect repo's main method for a specific set of inputs.
```
Comment (Leo-Neat) [inline infra/base-images/base-builder/detect_repo_test.py]: Confusing comment

File: infra/bisector.py [hunk 5]
```diff
@@ -119,18 +119,20 @@ def bisect(commit_old, commit_new, testcase, fuzz_target, build_data):
     commit_list = bisect_repo_manager.get_commit_list(commit_old, commit_new)
     old_idx = len(commit_list) - 1
     new_idx = 0
-
-    build_specified_commit.build_fuzzers_from_commit(build_data,
-                                                     commit_list[new_idx],
-                                                     bisect_repo_manager)
+    print(bisect_repo_manager)
```
Comment (Leo-Neat) [inline infra/bisector.py]: remove debug statement.

File: infra/bisector_test.py [hunk 6]
```diff
@@ -21,99 +21,39 @@
...
+
+  def test_bisect(self):
+    """Test the bisect method on example projects."""
+    for test_repo in test_repos.TEST_REPOS:
+      build_data = build_specified_commit.BuildData(
+          project_name=test_repo.project_name,
+          engine='libfuzzer',
+          sanitizer='address',
+          architecture='x86_64')
+      error_sha = bisector.bisect(test_repo.old_commit, test_repo.new_commit,
```
Comment (Leo-Neat) [inline infra/bisector_test.py]: change names to match old_commit vs. commit_old, testcase vs. test_case_path

File: infra/build_specified_commit_test.py [hunk 7]
```diff
@@ -22,6 +22,7 @@
 import build_specified_commit
 import helper
 import repo_manager
+import test_repos
 
 # Necessary because __file__ changes with os.chdir
```
Comment (Leo-Neat) [inline infra/build_specified_commit_test.py]: Delete

File: infra/test_repos.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+# the specified commits existing.
+TEST_REPOS = [
+    ExampleRepo(project_name='usrsctp',
+                oss_repo_name='usrsctp',
+                git_repo_name='usrsctp',
+                git_url='https://github.com/weinrank/usrsctp',
+                old_commit='4886aaa49fb90e479226fcfc3241d74208908232',
+                new_commit='c710749b1053978179a027973a3ea3bccf20ee5c',
+                intro_commit='457d6ead58e82584d9dcb826f6739347f59ebd3a',
+                fuzzer='fuzzer_connect',
```
Comment (Leo-Neat) [inline infra/test_repos.py]: change to fuzz target.

File: infra/base-images/base-builder/detect_repo_test.py (hunk 1 above)
Comment (Leo-Neat) [reply infra/base-images/base-builder/detect_repo_test.py]: Done.

File: infra/base-images/base-builder/detect_repo_test.py (hunk 2 above)
Comment (Leo-Neat) [reply infra/base-images/base-builder/detect_repo_test.py]: Done.

File: infra/base-images/base-builder/detect_repo_test.py (hunk 3 above)
Comment (Leo-Neat) [reply infra/base-images/base-builder/detect_repo_test.py]: Done.

File: infra/base-images/base-builder/detect_repo_test.py (hunk 4 above)
Comment (Leo-Neat) [reply infra/base-images/base-builder/detect_repo_test.py]: Done.

File: infra/bisector.py (hunk 5 above)
Comment (Leo-Neat) [reply infra/bisector.py]: Done.

File: infra/test_repos.py (hunk 8 above)
Comment (Leo-Neat) [reply infra/test_repos.py]: Done.

File: infra/bisector_test.py (hunk 6 above)
Comment (Leo-Neat) [reply infra/bisector_test.py]: Done.

**Comment by Dor1s** [comment]
> Updating some of the unit and integration tests for OSS-Fuzz infrastructure.

"for CIFuzz infrastructure" sounds more correct IMO

File: infra/base-images/base-builder/detect_repo_test.py [hunk 9]
```diff
@@ -30,94 +32,52 @@
...
       tmp_dir: The location of the directory of git repos to be searched.
+      commit: The commit that should be used to look up the repo.
     """
-    command = [
-        'python3', 'detect_repo.py', '--src_dir', tmp_dir, '--example_commit',
-        commit
-    ]
+    command = ['python3', 'detect_repo.py', '--src_dir', tmp_dir]
+
+    if not commit:
```
Comment (oliverchang) [inline infra/base-images/base-builder/detect_repo_test.py]: nit: invert conditions :)

Comment (Leo-Neat) [reply infra/base-images/base-builder/detect_repo_test.py]: Fixed

**Pull request merged**